src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,91 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75F;
+    private static final int GROW_COEFFICIENT = 2;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    private class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+            hash = key == null ? 0 : Math.abs(key.hashCode()) % table.length;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+
+    }
 
     @Override
     public void put(K key, V value) {
-
+        checkForResize();
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[newNode.hash] == null) {
+            table[newNode.hash] = newNode;
+        } else {
+            Node<K, V> temporaryNode = table[newNode.hash];
+            do {
+                if (Objects.equals(key, temporaryNode.key)) {
+                    temporaryNode.value = newNode.value;
+                    return;
+                }
+                if (temporaryNode.next == null) {
+                    temporaryNode.next = newNode;
+                    break;
+                }
+                temporaryNode = temporaryNode.next;
+            } while (temporaryNode != null);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> temporaryNode = table[getHashCodeFromKey(key)];
+        while (temporaryNode != null) {
+            if (Objects.equals(key, temporaryNode.key)) {
+                return temporaryNode.value;
+            }
+            temporaryNode = temporaryNode.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void checkForResize() {
+        if (size == threshold) {
+            Node<K, V>[] temporaryTable = table;
+            table = new Node[table.length * GROW_COEFFICIENT];
+            size = 0;
+            for (Node<K, V> node : temporaryTable) {
+                while (node != null) {
+                    put(node.key, node.value);
+                    node = node.next;
+                }
+            }
+            threshold = (int) (table.length * DEFAULT_LOAD_FACTOR);
```

Comment:
maybe put table.langth to variable in class MyHashMap so  you don't need to calculate it each time